### PR TITLE
Make GuaranteedThroughputSampler public

### PR DESCRIPTION
**Pull request by quaff**

### CODE REVIEW

**Comment by jpkrohling** [comment]
@quaff, do you have a reason to require this to be public? Also, looks like your commit is missing the DCO sign-off.

File: jaeger-core/src/main/java/io/jaegertracing/samplers/GuaranteedThroughputSampler.java [hunk 1]
```diff
@@ -31,15 +30,15 @@
  */
 @ToString
 @EqualsAndHashCode
-@AllArgsConstructor (access = AccessLevel.PACKAGE) // Visible for testing
+@AllArgsConstructor
```
Comment (jpkrohling) [inline jaeger-core/src/main/java/io/jaegertracing/samplers/GuaranteedThroughputSampler.java]: Just noticed this change here: this exposes a *new* constructor, to initialize the private fields (`probabilisticSampler`, `lowerBoundSampler` and `tags`). 

I don't think we want that. In fact, I think this annotation can be removed altogether.

**Comment by quaff** [comment]
@jpkrohling commit updated, since ProbabilisticSampler and RateLimitingSampler are public, why not GuaranteedThroughputSampler?

**Comment by jpkrohling** [comment]
@quaff Looks like a recent change caused merge conflicts with this PR. Would you please update it? Once it's rebased to the latest master, I'll merge it.

**Comment by quaff** [comment]
@jpkrohling updated.

**Comment by jpkrohling** [comment]
Thanks, @quaff!

**Pull request merged**